addons/stock_dropshipping/models/sale.py

```diff
@@ -49,3 +49,11 @@ def _get_qty_procurement(self, previous_product_uom_qty):
             return qty
         else:
             return super(SaleOrderLine, self)._get_qty_procurement(previous_product_uom_qty=previous_product_uom_qty)
+
+    @api.depends('purchase_line_count')
+    def _compute_product_updatable(self):
+        super()._compute_product_updatable()
+        if self.env.user.has_group('purchase.group_purchase_user'):
+            for line in self:
+                if line.purchase_line_count > 0:
+                    line.product_updatable = False
```

Comment:
I suppose it wouldn't be a bad idea to move this directly in `sale_purchase` :thinking:.
When would we want to modify the product of a SOL with POL ?

Reply:
Agreed that this is probably the case.
Thought it's more the expertise of the stock team, I focus on keeping the same behavior than before as much as possible.

So for the moment I leave it like that, and warn the stock team there is probably something to improve here